src/controllers/auth.controller.js

```diff
@@ -0,0 +1,269 @@
+  const { activationToken } = req.params;
+  const user = await User.findOne({ where: { activationToken } });
+
+  if (!user) {
+    res.sendStatus(404);
+
+    return;
+  }
+
+  user.activationToken = null;
+  await user.save();
+
+  res.send(user);
+};
+
+const login = async (req, res) => {
+  const { email, password } = req.body;
+
+  const user = await userService.findByEmail(email);
+
+  if (!user) {
+    throw ApiError.badRequest('Invalid email or password');
+  }
+
+  const isPasswordValid = await bcrypt.compare(password, user.password);
+
+  if (!isPasswordValid) {
+    throw ApiError.badRequest('Invalid email or password');
+  }
+
+  await generateToken(res, user);
+};
+
+const refresh = async (req, res) => {
+  const { refreshToken } = req.cookies;
+
+  const userData = jwtService.verifyRefresh(refreshToken);
+  const token = await tokenService.getByToken(refreshToken);
+
+  if (!userData || !token) {
+    throw ApiError.unauthorized();
+  }
+
+  const user = await userService.findByEmail(userData.email);
+
+  await generateToken(res, user);
+};
+
+const logout = async (req, res) => {
+  const { refreshToken } = req.cookies;
+  const userData = await jwtService.verifyRefresh(refreshToken);
+
+  if (!userData || !refreshToken) {
+    throw ApiError.unauthorized();
+  }
+
+  await tokenService.remove(userData.id);
+
+  res.sendStatus(204);
+};
+
+const passwordReset = async (req, res) => {
+  const { email } = req.body;
+  const user = await userService.findByEmail(email);
+
+  if (!user) {
+    throw ApiError.unauthorized('User with that email does not exist');
+  }
+
+  const resetToken = jwtService.sign(userService.normalize(user));
+
+  await tokenService.save(user.id, resetToken);
+  await sendResetEmail.sendResetEmail(user.email, resetToken);
+
+  res.send({ message: 'Password reset link sent to your email' });
+};
+
+const resetPassword = async (req, res) => {
+  const { token, password, confirmPassword } = req.body;
+
+  if (password !== confirmPassword) {
+    throw ApiError.badRequest('Passwords do not match');
+  }
+
+  const userData = jwtService.verify(token);
+
+  if (!userData) {
+    throw ApiError.badRequest('Invalid or expired token');
+  }
+
+  const user = await User.findByPk(userData.id);
+
+  if (!user) {
+    throw ApiError.badRequest('User not found');
+  }
+
+  const hashedPass = await bcrypt.hash(password, 10);
+
+  user.password = hashedPass;
+
+  await user.save();
+
+  res.send({ message: 'Password successfully reset' });
+};
+
+const updateProfile = async (req, res) => {
+  const {
+    name,
+    oldPassword,
+    newPassword,
+    confirmPassword,
+    newEmail,
+    confirmEmail,
+  } = req.body;
+  const userId = req.user.id;
+
+  const user = await User.findByPk(userId);
+
+  if (!user) {
+    throw ApiError.notFound('User not found');
+  }
+
+  if (name) {
+    if (name.trim().length < 2) {
+      throw ApiError.badRequest('Name should be at least 2 characters long');
```

Comment:
Consider using `return next(ApiError.badRequest('Name should be at least 2 characters long'));` instead of `throw` to ensure the error is passed to the next middleware for consistent error handling.